### ISSUE TITLE
Popup  layout is skewed.

### ISSUE THREAD

**Issue by YugoMatsuda** [iDevelopper/PBPopupController#12]
Thank you for amazing library.
It's been pretty helpful for me!!

But when I get back app after background, the layout is skewed like a Photo.
![IMG_4289](https://user-images.githubusercontent.com/61716314/103458744-64458e80-4d4e-11eb-907c-4900988d266c.PNG)

I’m using a code as specified in the documentation like below
Do I need to write anything else?
Thanks!

   self.tabBarController?.popupController.delegate = self
        
        if let popupBar = self.tabBarController?.popupBar {

            popupBar.dataSource = self
            
            popupBar.image = UIImage(named: "Cover01")
            popupBar.title = "Title"
            popupBar.subtitle = " A subtitle"
            popupBar.accessibilityLabel = "My custom label"
            
            let popupPlayButtonItem = UIBarButtonItem(image: #imageLiteral(resourceName: "play-small"), style: .plain, target: self, action: #selector(playPauseAction(_:)))
            popupPlayButtonItem.accessibilityLabel = NSLocalizedString("Play", comment: "")
            let popupNextButtonItem = UIBarButtonItem(image: #imageLiteral(resourceName: "next-small"), style: .plain, target: self, action: #selector(nextAction(_:)))
            popupNextButtonItem.accessibilityLabel = NSLocalizedString("Next track", comment: "")
            
            popupBar.rightBarButtonItems = [popupPlayButtonItem, popupNextButtonItem]
            
            let popupContentVC = self.storyboard?.instantiateViewController(withIdentifier: "PopupContentViewController") as? PopupContentViewController
            
            self.tabBarController?.presentPopupBar(withPopupContentViewController: popupContentVC, animated: true, completion: {
                print("Presented")
            })
        }
.

**Comment by iDevelopper** [comment]
Not sure to understand. Can you provide a sample project please?

Thanks,

**Comment by YugoMatsuda** [comment]
> Not sure to understand. Can you provide a sample project please?
> 
> Thanks,

Thank you for reply.
I've upploaded sample minimum project to my repositry.
Please confirm.
https://github.com/YugoMatsuda/SmaplePBPopup

**Comment by YugoMatsuda** [comment]
If you detect the cause of that Please tell me.
Thanks.

**Comment by iDevelopper** [comment]
Try like that:
```swift
    @IBAction func displayPopupAction(_ sender: Any) {
        self.tabBarController?.popupController.delegate = self
        self.tabBarController?.popupContentView.popupIgnoreDropShadowView = true
        if let popupBar = self.tabBarController?.popupBar {
            
            popupBar.dataSource = self
            popupBar.title = "Title"
            
            let popupContentVC = PopUpViewController()
            popupContentVC.view.backgroundColor = .gray
            
            self.tabBarController?.presentPopupBar(withPopupContentViewController: popupContentVC, animated: true, completion: {
                print("Presented")
            })
        }
    }
```

**Comment by YugoMatsuda** [comment]
It worked!!
Thank you so much!!

**Comment by iDevelopper** [comment]
Nothing, keep me informed.

**Comment by YugoMatsuda** [comment]
Sorry, I have a question!
I displayed mini player popUp many times, then If I display popUpContentController, screen is stuck.
Is there any way to avoid this?
Thanks.

**Comment by iDevelopper** [comment]
Can you elaborate please?
I cannot reproduce this behavior with your example.

> I displayed mini player popUp many times

What do you mean by many times?

> If I display popUpContentController

How do you present it? By user gesture?


Can you upload a new sample project?

**Comment by YugoMatsuda** [comment]
Thank you for reply.
I've reproduced issue, comiiited sample minimum project to same repositry.
https://github.com/YugoMatsuda/SmaplePBPopup

When I tapped Display Popup Button many times in a row, then  If I tapped popUp content, 
popUpContentController display slowly.
popUpContentController's movement will be slow.

What should I do??
Please confirm.

**Comment by iDevelopper** [comment]
You can decrease the presentation and closing duration of the popup content view with the popupPresentationDuration (default is 0.5) and popupDismissalDuration (default is 0.6) properties.

example:
```swift
        self.tabBarController?.popupController.delegate = self
        self.tabBarController?.popupContentView.popupIgnoreDropShadowView = true
        self.tabBarController?.popupContentView.popupPresentationDuration = 0.3
        self.tabBarController?.popupContentView.popupDismissalDuration = 0.4
```
Is that your question?

**Comment by YugoMatsuda** [comment]
No,,,,
PopUpContentController Screen transition moves slower as I tapped display button many times.
like below video.
Thanks.

https://user-images.githubusercontent.com/61716314/104118244-e1828c00-536a-11eb-819b-81aaa0bc2af1.MP4

**Comment by iDevelopper** [comment]
This seems to be an Apple iOS bug (allocations are increasing) as I don't see anything abnormal regarding the framework in Instrument and in the view memory graph hierarchy.

For now try this:
```swift
            if self.tabBarController?.popupController.popupPresentationState == .hidden {
                self.tabBarController?.presentPopupBar(withPopupContentViewController: popupContentVC, animated: true, completion: {
                    print("Presented")
                })
            }
```

**Comment by YugoMatsuda** [comment]
Okay.
I appreciate your cooperation. 
ありがとうございました！

**Comment by iDevelopper** [comment]
Thank you for having pointed this.

Normally we only present the popup bar once but fixed with version 3.4.1.

If you have another problem, please open a new issue.

Can you please test without the last check:
```swift
            //if self.tabBarController?.popupController.popupPresentationState == .hidden {
                self.tabBarController?.presentPopupBar(withPopupContentViewController: popupContentVC, animated: true, completion: {
                    print("Presented")
                })
            //}
```


Chers,
